packages/router/src/page-loader.js

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react'
+
+import { createNamedContext } from './internal'
+
+const PageLoadingContext = createNamedContext('PageLoading')
+
+const PageLoader = ({ spec, delay, params }) => {
+  const [cache, setCache] = useState({})
+  const [pageName, setPageName] = useState(null)
+  const [loading, setLoading] = useState(false)
+
+  const { loader, name } = spec
+
+  let loadingTimeout
```

Comment:
I think we should store the loadingTimeout as a ref, with [useRef](https://reactjs.org/docs/hooks-reference.html#useref), that way we can guarantee that we get the same value on every render.